sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -731,3 +732,11 @@ case class HiveTableRelation(
     s"$nodeName $metadataStr"
   }
 }
+
+/**
+ * A V2 table with V1 fallback support. This is used to fallback to V1 table when the V2 one
+ * doesn't implement specific capabilities but V1 already has.
+ */
+trait V2TableWithV1Fallback extends Table {
```

Comment:
shall we define it in the same file as `V1Table`?

Reply:
Copy, moved in 97761d2